tests/integration/commits/test_commits.py

```diff
@@ -0,0 +1,122 @@
+        credentials_response = mock.Mock()
+        credentials_response.json.return_value = load_json_fixture('katka_credentials.json')
+        mock_credentials_client.get.return_value = credentials_response
+
+        bitbucket_response = mock.Mock()
+        bitbucket_response.json.return_value = load_json_fixture('bitbucket_no_commits.json')
+        mock_bitbucket_request.return_value = bitbucket_response
+
+        endpoint = reverse('commits', kwargs={'project_key': 'msap', 'repository_name': 'invisible_commits'})
+        response = client.get(
+            endpoint,
+            content_type='application/json',
+            data={'credential': '3617e084-950c-4a44-aa6d-b4aeba1db428', 'limit': 10, 'start': 2}
+        )
+
+        assert response.status_code == 200
+        assert len(response.data['values']) == 0
+        assert response.data['start'] == 0
+        assert response.data['limit'] == 0
+        assert response.data['size'] == 0
+        assert response.data['is_last_page'] is True
+        assert response.data['next_page_start'] == 0
+
+    def test_bitbucket_exception(self, mock_credentials_client, mock_bitbucket_request, load_json_fixture, client):
+        """ If the token is not valid bitbucket will retrieve an empty list of projects """
+        mock_credentials_client.return_value = mock.PropertyMock()
+        credentials_response = mock.Mock()
+        credentials_response.json.return_value = load_json_fixture('katka_credentials.json')
+        mock_credentials_client.get.return_value = credentials_response
+
+        bitbucket_response = mock.Mock()
+        bitbucket_response.raise_for_status.side_effect = HTTPError(
+            'test exception', response=mock.Mock(content=None, status_code=500)
+        )
+        mock_bitbucket_request.return_value = bitbucket_response
+
+        endpoint = reverse('commits', kwargs={'project_key': 'msap', 'repository_name': 'invisible_women'})
+        response = client.get(
+            endpoint,
+            content_type='application/json',
+            data={'credential': '4cb432f5-0def-48b6-ad05-f1c082b1f1b9', 'limit': 10, 'start': 2}
+        )
+
+        assert response.status_code == BitbucketBaseAPIException.status_code
+
+
+class TestGetProjectsNoCredentials:
+    @mock.patch('requests.sessions.Session.request')
+    def test_no_credential(self, mock_bitbucket_request, load_json_fixture, client):
+        bitbucket_response = mock.Mock()
+        bitbucket_response.json.return_value = load_json_fixture('bitbucket_no_commits.json')
+        mock_bitbucket_request.return_value = bitbucket_response
+
+        endpoint = reverse('commits', kwargs={'project_key': 'msap', 'repository_name': 'invisible_commits'})
+        response = client.get(
+            endpoint,
+            content_type='application/json',
+            data={'limit': 10, 'start': 2}
+        )
+
+        assert response.status_code == 200
```

Comment:
Why is the response 200 if you get without credentials?

Reply:
Because there are repos that are public so u don't need a credential to access it.